typespec-extension/src/emitter.ts

```diff
@@ -101,14 +104,19 @@ export async function $onEmit(context: EmitContext<EmitterOptions>) {
     const jarFileName = resolvePath(moduleRoot, "target", "azure-typespec-extension-jar-with-dependencies.jar");
     program.trace("typespec-java", `Exec JAR ${jarFileName}`);
 
+    const javaArgs: string[] = [];
+    javaArgs.push(`-DemitterOptions=${emitterOptions}`);
+    if (options["dev-options"]?.debug) {
+      javaArgs.push("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005")
+    }
+    if (options["dev-options"]?.loglevel) {
+      javaArgs.push("-Dloglevel=" + options["dev-options"]?.loglevel);
+    }
+    javaArgs.push("-jar");
+    javaArgs.push(jarFileName);
+    javaArgs.push(codeModelFileName);
     try {
-      const output = await promisify(execFile)("java", [
-        `-DemitterOptions=${emitterOptions}`,
-        "-jar",
-        jarFileName,
-        codeModelFileName,
-      ]);
-      program.trace("typespec-java", output.stdout ? output.stdout : output.stderr);
+      spawn("java", javaArgs, {stdio: "inherit"});
```

Comment:
There seems to be some difference of async of the change.

After `promisify` I assume it be a `Promise`. Then `await` on it, as `$onEmit` is async function.

But `spawn` seems to not be a `Promise`.
https://nodejs.org/api/child_process.html#child_processspawncommand-args-options

Reply:
The problem about it, is that `$onEmit` would finish (and tsp would continue), regardless of this java call is success or not.

Reply:
Have updated the code to handle exit and error events and notify the emitter of any errors from running the java process.